File: Leetcode/53. Maximum Subarray.ts
function maxSubArray(nums: number[]): number {
    let mth: number = nums[0], msf: number = nums[0];
    
    for(let i=1; i<nums.length; i++) {
        mth += nums[i];
        if(mth <= nums[i]) mth = nums[i];
        if(msf < mth) msf = mth;
    }
    
    return msf;
};

File: Leetcode/C++/53. Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int mx = nums[0], cur = nums[0];
        
        for(int i=1; i<nums.size(); i++) {
            cur += nums[i];
            if(cur <= nums[i]) cur = nums[i];
            if(cur > mx) mx = cur;
        }
        
        return mx;
    }
};
